src/Core/Framework/DataAbstractionLayer/Search/EntitySearchResult.php

```diff
@@ -84,18 +61,14 @@ public function getContext(): Context
         return $this->context;
     }
 
-    public function clear(): void
+    public function getLimit(): ?int
     {
-        parent::clear();
-
-        $this->entities->clear();
+        return $this->limit;
     }
 
-    public function add($entity): void
+    public function getPage(): int
     {
-        parent::add($entity);
-
-        $this->entities->add($entity);
+        return $this->page;
     }
 
     public function jsonSerialize(): array
```

Comment:
if we do not want to use the trait, we should at least implement the JsonSerializable interface right? :sweat_smile: